assets/js/googlesitekit/data/create-fetch-store.js

```diff
@@ -0,0 +1,270 @@
+
+	const fetchCreator = `fetch${ pascalCaseBaseName }`;
+	const receiveCreator = `receive${ pascalCaseBaseName }`;
+	const isFetching = `isFetching${ pascalCaseBaseName }`;
+
+	const INITIAL_STATE = {
+		[ isFetching ]: {},
+	};
+
+	const actions = {
+		*[ fetchCreator ]( ...args ) {
+			let response;
+			let error;
+			let params;
+
+			try {
+				params = argsToParams( ...args );
+			} catch ( err ) {
+				// Parameters should never be invalid here, this needs to be
+				// strict and inform the developer of the issue.
+				global.console.error( err.message );
+				error = err;
+				return { response, error };
+			}
+
+			yield {
+				payload: { params },
+				type: START_FETCH,
+			};
+
+			try {
+				response = yield {
+					payload: { params },
+					type: FETCH,
+				};
+
+				yield actions[ receiveCreator ]( response, params );
+
+				yield {
+					payload: { params },
+					type: FINISH_FETCH,
+				};
+			} catch ( e ) {
+				error = e;
+
+				yield {
+					payload: { error, params },
+					type: CATCH_FETCH,
+				};
+			}
+
+			return { response, error };
+		},
+
+		[ receiveCreator ]( response, params ) {
+			invariant( response !== undefined, 'response is required.' );
+			if ( requiresParams ) {
+				invariant( isPlainObject( params ), 'params is required.' );
+			} else {
+				params = {};
+			}
+
+			return {
+				payload: { response, params },
+				type: RECEIVE,
+			};
+		},
+	};
+
+	const controls = {
+		[ FETCH ]: ( { payload } ) => {
+			return controlCallback( payload.params );
+		},
+	};
+
+	const reducer = ( state, { type, payload } ) => {
+		switch ( type ) {
+			case START_FETCH: {
+				const { params } = payload;
+				return {
+					...state,
+					[ isFetching ]: {
+						...state[ isFetching ],
+						[ stringifyObject( params ) ]: true,
+					},
+				};
+			}
+
+			case RECEIVE: {
+				const { response, params } = payload;
+				return reducerCallback( state, response, params );
+			}
+
+			case FINISH_FETCH: {
+				const { params } = payload;
+				return {
+					...state,
+					[ isFetching ]: {
+						...state[ isFetching ],
+						[ stringifyObject( params ) ]: false,
+					},
+				};
+			}
+
+			case CATCH_FETCH: {
+				const { error, params } = payload;
+				return {
+					...state,
+					error,
+					[ isFetching ]: {
+						...state[ isFetching ],
+						[ stringifyObject( params ) ]: false,
+					},
+				};
+			}
+
+			default: {
+				return { ...state };
+			}
+		}
+	};
+
+	const selectors = {
+		[ isFetching ]: ( state, ...args ) => {
+			if ( state[ isFetching ] === undefined ) {
+				return false;
+			}
+
+			let params;
+			try {
+				params = argsToParams( ...args );
+			} catch ( err ) {
+				// If parameters are invalid, fail silently here. It likely is
+				// because some dependency selector is still resolving.
+				return false;
+			}
+
+			return !! state[ isFetching ][ stringifyObject( params ) ];
+		},
+	};
+
+	return {
+		INITIAL_STATE,
+		actions,
+		controls,
+		reducer,
+		resolvers: {},
```

Comment:
Technically we can omit this key rather than pass an empty object, right? Not a big deal either way.

Reply:
I think it's better to include it so that the function returns a full store. We're using "partial store objects" wherever we combine them with `Data.combineStores` (so that it wouldn't matter), but any store object that gets exported or returned from a function should be complete in itself I'd argue.